### PR TITLE
queue system and ve optimisations

### PR DESCRIPTION
**Pull request by 20dka**

### CODE REVIEW

File: lua/ge/extensions/MPGameNetwork.lua [hunk 1]
```diff
@@ -107,8 +107,8 @@ local HandleNetwork = {
 	['Z'] = function(params) positionGE.handle(params) end,
 	['O'] = function(params) MPVehicleGE.handle(params) end,
 	['P'] = function(params) MPConfig.setPlayerServerID(params) end,
-	['J'] = function(params) onPlayerConnect() UI.showNotification(params) end, -- A player Joined
-	['L'] = function(params) UI.showNotification(params) end, -- A player Joined
+	['J'] = function(params) onPlayerConnect() UI.showNotification(params) end, -- A player joined
+	['L'] = function(params) UI.showNotification(params) end, -- A player left
```
Comment (Starystars67) [inline lua/ge/extensions/MPGameNetwork.lua:111]: This is not `A player left` it should be something like `Display custom notification`

**Review by Starystars67 — COMMENTED** [review]
This does look good to me. Have we confirmed that it is all working code first?
I do not have time to load this PR in just to test at the moment, sorry.

**Comment by 20dka** [comment]
Ive found a potential bug when loading into a server (not all vehicles load until the car youre spectating stops) but other than that i think this is ready to be pushed to EA.
We do have some community members who are waiting for certain patches in this PR

**Comment by jojos38** [comment]
> Ive found a potential bug when loading into a server (not all vehicles load until the car youre spectating stops) but other than that i think this is ready to be pushed to EA.
> We do have some community members who are waiting for certain patches in this PR

Let's fix this issue before releasing then, I can help you this evening if you want

**Comment by 20dka** [comment]
i might be busy tonight

**Comment by 20dka** [comment]
this PR has been open for so long that now it has a conflict.. could we please look into merging this?

**Pull request merged**